tools/codegen/api/autograd.py

```diff
@@ -259,27 +267,53 @@ def find_info(f: NativeFunction) -> Tuple[Optional[DifferentiabilityInfo], bool]
             if f.func.kind() == SchemaKind.inplace:
                 assert len(info.forward_derivatives) == 1  # Only single output inplace should exist
                 fw_info = info.forward_derivatives[0]
-
-                if re.search(IDENT_REGEX.format("self"), fw_info.formula):
-                    raise RuntimeError(f'The formula for "{f.func.name}" is using the original value of self that is being '
-                                       'modified inplace. This would lead to wrong forward gradients. Please use "result" in '
-                                       'the formula only.')
-
-                # replace "result" from the formula by self
+                formula = fw_info.formula
+
+                def update_for_original(formula: str, postfix: str) -> str:
```

Comment:
nit: I had to read the use sites to figure out what this is doing. How about calling this "replace_self_with_original_self"?